profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.blocks.inc

```diff
@@ -0,0 +1,889 @@
+      'content' => '',
+    ),
+  );
+  $num_columns = variable_get('newsroom_summary_block_columns', 1);
+
+  // Sort boxes based on configuration.
+  uasort($boxes, '_nexteuropa_newsroom_sort_newsroom_boxes');
+
+  $newsroom_row_number = 1;
+  foreach ($boxes as $box) {
+    $active_column = 1;
+    $box['classes'][] = 'newsroom_item_container';
+    // Get active column.
+    if ($num_columns == 2 && $newsroom_row_number % 2 == 0) {
+      $active_column = 2;
+    }
+
+    $columns_newsroom[$active_column]['visible'] = TRUE;
+    $columns_newsroom[$active_column]['content'] .= '<div id="' . $box['id'] . '" class="' . implode(' ', $box['classes']) . '">'; /* add "highlighted" classname to display that block as highlighted (ie press releases, speeches) */
+    $columns_newsroom[$active_column]['content'] .= '<a name="' . $box['id'] . '"></a>';
+    $columns_newsroom[$active_column]['content'] .= '<h3 class="newsroom_title">' . $box['title'] . '</h3>';
+    $columns_newsroom[$active_column]['content'] .= $box['block']['content'];
+    $columns_newsroom[$active_column]['content'] .= '<div class="newsroom_more">' . $box['list_link'] . '</div>';
+    $columns_newsroom[$active_column]['content'] .= "</div>";
+
+    $newsroom_row_number++;
+  }
+
+  // 2 columns so display both.
+  if ($columns_newsroom[2]['visible'] === TRUE) {
+    $newsroom_content .= '<div class="nr_column_1 nr_column">' . $columns_newsroom[1]['content'] . '</div>';
+    $newsroom_content .= '<div class="nr_column_2 nr_column">' . $columns_newsroom[2]['content'] . '</div>';
+  }
+  // 1 column only.
+  else {
+    $newsroom_content .= '<div class="nr_column_single">' . $columns_newsroom[1]['content'] . '</div>';
+  }
+  return $newsroom_content;
+}
+
+/**
+ * Sort newsroom boxes.
+ *
+ * @param array $a
+ *   Newsroom box.
+ * @param array $b
+ *   Newsroom box.
+ *
+ * @return bool
+ *   Value.
+ */
+function _nexteuropa_newsroom_sort_newsroom_boxes(array $a, array $b) {
+  // Highligths at begining or end - based on setting.
+  if ($a['highlight'] !== $b['highlight']) {
+    return variable_get('newsroom_display_highlights_begin', 0) ? $a['highlight'] == FALSE : $a['highlight'] == TRUE;
+  }
+  // Sort Agenda before or after highlighted  - based on setting.
+  if ($a['highlight'] && ($a['agenda'] !== $b['agenda'])) {
+    return variable_get('newsroom_agenda_after_highlights', 0) ? $a['agenda'] == TRUE : $a['agenda'] == FALSE;
+  }
+  // Sort based on last update  - based on setting.
+  if (variable_get('newsroom_order_by_last_update', 0)) {
+    return $a['last_update'] < $b['last_update'];
+  }
+  else {
+    return $a['weight'] > $b['weight'];
+  }
+}
+
+/**
+ * Helper to check if topic is highlighted.
+ *
+ * @return bool
+ *   is highlighted or not
+ */
+function _nexteuropa_newsroom_type_is_highlighted($term) {
+  if ($term === NEXTEUROPA_NEWSROOM_CALENDAR_BLOCK) {
+    return FALSE;
+  }
+  $item_wrapper = entity_metadata_wrapper('taxonomy_term', $term);
```

Comment:
You don't need emw here to have the value, please don't create possible performance issues.